agent/engine/docker_image_manager.go

```diff
@@ -277,6 +275,8 @@ func (imageManager *dockerImageManager) performPeriodicImageCleanup(ctx context.
 }
 
 func (imageManager *dockerImageManager) removeUnusedImages() {
+	imageManager.updateLock.Lock()
```

Comment:
Can you document if you've observed any slowness because of this in task launch latencies on an instance where there are at least 5 images to clean?